### PR TITLE
BLD: Switch to versioneer

### PR DESCRIPTION
**Pull request by bashtage**
Simply release by using versioneer to handle versioning
Add a v0.10.0dev tag to generate correct build numbers from master

### CODE REVIEW

**Comment by TomAugspurger** [comment]
>  Add a v0.10.0dev tag to generate correct build numbers from master

The tag doesn't come through in the PR, correct? It'll have to be pushed to master?

And it's important to use `v0.10.0.dev0` (with the last 0), else you'll get warnings that `0.10.0dev` isn't a valid version.

**Comment by bashtage** [comment]
Are you sure?  I didn't see any warnings:

https://api.travis-ci.org/v3/job/381632573/log.txt

Versioneer seems to automatically produce 0.10.0.dev0+COMMIT since there are no commits past the tag.

**Comment by TomAugspurger** [comment]
No, not 100%. I just recall a pandas issue where we tagged it with `.dev`
instead of `.dev0` and got warnings. Things could have changed.

**Comment by josef-pkt** [comment]
This pull request **introduces 2 alerts** when merging ca75ab15eafc213d079bf5bb436a7f52f9706f90 into ec3b6d02c96cd9c8f4b993434f0bbae4b3e91a21 - [view on lgtm.com](https://lgtm.com/projects/g/statsmodels/statsmodels/rev/pr-e020d942b24a37f7a2a8043d007c43e4c1dfe390)

**new alerts:**

* 1 for Suspicious unused loop iteration variable
* 1 for Variable defined multiple times

---

*Comment posted by [lgtm.com](https://lgtm.com)*

File: setup.cfg [hunk 1]
```diff
@@ -0,0 +1,7 @@
+[versioneer]
```
Comment (jbrockmendel) [inline setup.cfg]: +1 for introducing this file.

File: setup.py [hunk 2]
```diff
@@ -332,7 +333,7 @@ def finalize_options(self):
     def run(self):
         pass
 
-
+cmdclass = versioneer.get_cmdclass()
 cmdclass = {'clean': CleanCommand,
```
Comment (jbrockmendel) [inline setup.py]: does it matter that `cmdclass` is being overwritten immediately?

**Comment by josef-pkt** [comment]
This pull request **introduces 1 alert** when merging b8f35c38fe679a2e14cb04bc0364660a1aeee6e6 into 6611e878fb625e20b587a34c208f69e1b9d2495b - [view on lgtm.com](https://lgtm.com/projects/g/statsmodels/statsmodels/rev/pr-44cce9524b9f64d154efa53698ba6ed63404411e)

**new alerts:**

* 1 for Suspicious unused loop iteration variable

---

*Comment posted by [lgtm.com](https://lgtm.com)*

**Comment by bashtage** [comment]
@jbrockmendel thanks.

File: statsmodels/__init__.py [hunk 3]
```diff
@@ -55,3 +55,7 @@ def __call__(self, extra_args=None):
     from .version import version as __version__
 except ImportError:
     __version__ = 'not-yet-built'
+
```
Comment (jbrockmendel) [inline statsmodels/__init__.py]: should the lines above this be removed?

**Comment by josef-pkt** [comment]
This pull request **introduces 1 alert** when merging 0b7a7997033ebeecfbe73ed7091139dc9498bc99 into 41244374340ed1c798f8ae6992c4edf829e42c6e - [view on LGTM.com](https://lgtm.com/projects/g/statsmodels/statsmodels/rev/pr-68822bee1f43802081c1f593cf4074d70fc92671)

**new alerts:**

* 1 for Suspicious unused loop iteration variable

---

*Comment posted by [LGTM.com](https://lgtm.com)*

**Comment by bashtage** [comment]
Closing this in favor of #4871 which improves setup and adds versioneer.

**Pull request closed without merging**